src/main/java/com/github/wvengen/maven/proguard/ProGuardMojo.java

```diff
@@ -580,27 +603,34 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 			args.add("-injars");
 			StringBuilder filter = new StringBuilder(fileToString(inJarFile));
 			if ((inFilter != null) || (!addMavenDescriptor)) {
-				filter.append("(");
-				boolean coma = false;
+				List<String> filterList = new ArrayList<>();
 
 				if (!addMavenDescriptor) {
-					coma = true;
-					filter.append("!META-INF/maven/**");
+					filterList.add(MAVEN_DESCRIPTORS_FILTER);
 				}
 
 				if (inFilter != null) {
-					if (coma) {
-						filter.append(",");
-					}
-					filter.append(inFilter);
+					filterList.add(inFilter);
 				}
 
-				filter.append(")");
+				filter.append(createFilterString(filterList));
 			}
 			args.add(filter.toString());
 		}
 
 		if (includeDependency) {
+			List<String> dependencyInjarFilterList = new ArrayList<>();
+			if (!addManifest) {
+				dependencyInjarFilterList.add(MANIFEST_FILTER);
+			}
+			if (!addMavenDescriptor) {
+				dependencyInjarFilterList.add(MAVEN_DESCRIPTORS_FILTER);
+			}
+			if (inFilter != null) {
+				dependencyInjarFilterList.add(inFilter);
+			}
```

Comment:
Not sure if `inFilter` should be considered as well, or if a separate parameter should be introduced.